### ISSUE TITLE
Autofac v3.5.2 resolving generics bug

### ISSUE THREAD

**Issue by alenkralj** [autofac/Autofac#683]
Autofac v3.5.2 released via NuGet on Tuesday, August 12 2014 contains bug where resolving generics is not possible. Bug is fixed in commit 93a38f0c593928b1d094780d2a7c79bd on 29 Sep 2014 and it appears to be included in v3.5.2 after NuGet release, so fix is not visible to the NuGet users. Is it possible to open v3.5.3 and release it via NuGet?

**Comment by tillig** [comment]
Can you please post a repro or unit test of some code that doesn't work along with what you're expecting? [We have several unit tests around generics that were passing under 3.5.2 and continue to pass in 4.0.0.](https://github.com/autofac/Autofac/tree/master/test/Autofac.Test/Features/OpenGenerics)

If something does need to be fixed, it'll most likely have to wait until the Autofac 4.0.0 release, which we're in the middle of now.

**Comment by alenkralj** [comment]
Test "TheSameaceholderWithThreeGenericParametersTypeCanAppearMultipleTimesInTheService" will fail with Autofac release v3.5.2 from NuGet. Test will not fail however with Autofac release v3.5.2 from Github since there is commit after NuGet release of v3.5.2. Not sure how Autofac handles versions, but commits are added to v3.5.2 after NuGet release of v3.5.2. Maybe that is the main problem. Maybe there should be v3.5.3?

Basically nothing needs to be fixed, since fix already exists (as part of commit 93a38f0c593928b1d094780d2a7c79bd). It just isn't included in current stable version of Autofac on NuGet.

**Comment by tillig** [comment]
Ah, yes, I see it now.

There are some challenges with rolling back and publishing an old version that I'm not sure I have time to handle. For example, we have since moved our build server from MyGet to AppVeyor and I'm not super comfortable with publishing an official build off my local dev machine... which would mean I'd have to eithe re-set-up the old build server or fix the old build so it works on AppVeyor.

Unfortunately, you may have to wait for 4.0.0 if you're looking for a stable version with this fix. We're pretty far down the path of 4.0.0 and with the limited time we have to spend on it (and the huge number of integration libraries we're bringing forward for compatibility) rolling things back to patch what will very soon be the "old version" is probably not in the cards.

Since the fix is in the next version, I'm going to close this. If you're desperate for that fix in the 3.5.x line, you may need to fork and maintain your own build until 4.0.0 is released. Alternatively, you can try the 4.0.0 betas that we're publishing to NuGet now, which should have the fix in them.
